src/core/include/openvino/core/node_output.hpp

```diff
@@ -131,7 +131,7 @@ class OPENVINO_API Output<const Node> {
     /// \brief Constructs a Output, referencing the zeroth output of the node.
     /// \param node A `shared_ptr` to the node for the output handle.
     template <typename T>
-    Output(const std::shared_ptr<const T>& node) : Output(node ? node->get_default_output() : Output<const Node>()) {}
+    Output(const std::shared_ptr<const T>& node) : Output(node ? node->output(0) : Output<const Node>()) {}
```

Comment:
Add test to this change. 
From test perspectives it looks like it works same as before change, so why it is required?

Reply:
There're operations that would produce an exception if we use `get_default_output()` i.e. VariadicSplit or LSTMSequence as they have multiple outputs, hence no implicit conversion is available for such nodes. My change enables it.

Regarding the change in _src/plugins/intel_gpu/src/plugin/transformations/kv_cache_fusion.cpp_, I just decided to be explicit and pass Output where Output is expected. In fact, nothing changes as such.

Reply:
Added pretty simple tests. Any ideas for some other test scenarios?

Reply:
Analyzed, wrong change. Removing.

Reply:
Please revert this change.
`node->get_default_output()` was added here on purpose
Nodes with multiply output usually don't have "default output", developers should choose exact output depending on context.

Reply:
Maybe documentation should be updated instead.
It mentions to take 0th output but in fact will take default output if available.
If node has not default output then it will throw.

Reply:
Updated description